Quaver.Shared/Screens/Result/UI/ResultHitDifferenceGraph.cs

```diff
@@ -0,0 +1,305 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Quaver.API.Enums;
+using Quaver.API.Helpers;
+using Quaver.API.Maps.Processors.Scoring;
+using Quaver.API.Maps.Processors.Scoring.Data;
+using Quaver.API.Replays;
+using Quaver.API.Replays.Virtual;
+using Quaver.Shared.Assets;
+using Quaver.Shared.Config;
+using Quaver.Shared.Graphics.Notifications;
+using Quaver.Shared.Screens.Loading;
+using Quaver.Shared.Skinning;
+using Wobble.Graphics;
+using Wobble.Graphics.Sprites;
+using Wobble.Logging;
+
+namespace Quaver.Shared.Screens.Result.UI
+{
+    public class ResultHitDifferenceGraph : Sprite
+    {
+        /// <summary>
+        /// The score processor.
+        /// </summary>
+        private ScoreProcessor Processor { get; set; }
+
+        /// <summary>
+        /// Hit stats that we care about (only hits), also releases scaled down according to their window multiplier.
+        /// </summary>
+        private List<HitStat> FilteredStats { get; set; }
+
+        /// <summary>
+        /// Time of the first hit.
+        /// </summary>
+        private int EarliestHitTime { get; set; }
+
+        /// <summary>
+        /// Time of the last hit.
+        /// </summary>
+        private int LatestHitTime { get; set; }
+
+        public ResultHitDifferenceGraph(ScalableVector2 size, ResultScreen screen)
+        {
+            Tint = Color.Black;
+            Alpha = 0.2f;
+            Size = size;
+
+            Processor = GetScoreProcessor(screen);
+
+            var middle = (int) (Height / 2);
+            CreateMiddleLine(middle);
+
+            // Compute the values for positioning hit differences.
+            var largestJudgement = Processor.JudgementWindow.Values.Max();
+            float ConvertDifferenceToY(float hitDifference) =>
+                (largestJudgement - hitDifference) * (middle / largestJudgement);
+
+            CreateJudgementAreas(middle, ConvertDifferenceToY);
+            CreateEarlyLateText(largestJudgement);
+
+            // Exit early if there are no dots to draw.
+            if (Processor.Stats == null)
+            {
+                return;
+                // Make some fake scores for debugging.
```

Comment:
Remove this or create a method for it that we can call whenever we need to test.